### PR TITLE
RHTAPSRE-467: Support the configuration of provider URL

### PR DESCRIPTION
**Pull request by gbenhaim**
When onboarding self-hosted SCM PAC can't determine the the base url, so it should be provided explicitly.
Allow users to mention the provider (SCM) URL using an annotation on the component.

### CODE REVIEW

**Review by johnmcollier — APPROVED** [review]
Changes look fine to me. Though I will wait for @mmorhun to review before merging.

I wasn't aware this code was still in HAS, but looks like it's called from the build-service.

@mmorhun we should seriously consider moving this code to the build-service. It's only dependency within HAS is [prepare.go](https://github.com/redhat-appstudio/application-service/blob/main/gitops/prepare/prepare.go), which AFAICT, is not used within HAS at all.

**Comment by gbenhaim** [comment]
thanks for the review @johnmcollier. I fixed the formatting.

**Comment by gbenhaim** [comment]
@johnmcollier I didn't understand from the test logs if it failed because of my change or because of an infra issue/flaky test. Can you please help me debug it?

**Comment by johnmcollier** [comment]
@gbenhaim The e2e test failure seems to have been a flake that's resolved itself. I see some flakiness in our controller tests as well, so I've retested those to confirm

**Comment by mmorhun** [comment]
@gbenhaim why cannot we retrieve git provider URL from the Component source URL?

**Comment by gbenhaim** [comment]
@mmorhun I followed the configuration fields PAC provides. Since PAC doesn't retrieve the SCM host for self-hosted SCMs from the repository url, I don't think we should do it as well.

Nothing promises that the hostname of the SCM API server can be computed from the repository url, for example in a case where cloning is done through a proxy.

File: gitops/generate_build.go
Comment (mmorhun) [inline gitops/generate_build.go:93]: Are you sure that passing empty config instead of `nil` won't change PaC behavior?

**Review by mmorhun — APPROVED** [review]
LGTM

File: gitops/generate_build.go
Comment (gbenhaim) [reply gitops/generate_build.go]: I don't think it would make any difference but to be on the safe side I updated the PR and kept the default `gitProviderConfig` to be `nil`.

**Pull request merged**
